pkg/cloud/services/networking/router.go

```diff
@@ -165,6 +165,30 @@ INTERFACE_LOOP:
 	return nil
 }
 
+func (s *Service) DeleteRouter(network *infrav1.Network) error {
+	exists, err := s.existsRouter(network.Router.ID)
+	if err != nil {
+		return err
+	}
+	if exists {
```

Comment:
I think a guard close would be more idiomatic in go, i.e.:

```golang
if !exist {
    s.logger.Info("No router", "router", network.Router.ID)
    return nil
}
if network.Subnet == nil || network.Subnet.ID == "" {
    s.logger.V(4).Info("No need to remove router interface since no subnet exists.")
    return nil
}
_, err := routers.RemoveInterface(s.client, network.Router.ID, routers.RemoveInterfaceOpts{
    SubnetID: network.Subnet.ID,
}).Extract()
if err != nil {
    return fmt.Errorf("unable to remove router interface: %v", err)
}
s.logger.V(4).Info("Removed RouterInterface of Router", "id", network.Router.ID)

err = routers.Delete(s.client, network.Router.ID).ExtractErr(

```
(same for DeleteNetwork)